myTinyURL/convert/convert.py

```diff
@@ -4,26 +4,60 @@
 import random, string
 
 class Convert:
+    """This class is the implementation for url conversion
+    """
+
+    # the length of short_id
     CONST_LENGTH = 6
-    def long2short(httpurl):
-        short_id = Convert.gen()
-        record = URL(httpurl=httpurl, short_id=short_id)
+
+    def long2short(http_url):
+        """Convert given long url to a short url
+
+        Call gen_shortid generate a random short id.
+        Save in database.
+        Using SITE_URL generate response
+
+        Args:
+            http_url: a url we want to convert
+
+        Return:
+            The response(SITE_URL + short_id) we just create.
+        """
+        short_id = Convert.gen_shortid()
+        record = URL(http_url=http_url, short_id=short_id)
         record.save()
-        response = settings.SITE_URL + "/" + short_id
+        response = settings.SITE_URL + '/' + short_id
         return response
 
     def short2long(short_id):
+        """Convert given short_id to its corresponding long url
+
+        Find in the database, the primary key is short_id
+        If found, update this record click counts and save it.
+
+        Args:
+            short_id: a short url we want to convert
+
+        Return:
+            The corresponding long url.
+        """
         url = get_object_or_404(URL, pk=short_id)
         url.count += 1
         url.save()
-        return url.httpurl
+        return url.http_url
+
+    def gen_shortid():
+        """Generate short_id randomly
+
+        Generate short_id randomly.
+        Check in the database, if this short_id doesn't occur, return it. Otherwise, generate another one.
```

Comment:
Check if the short_id exists in the database. Return if it does. Otherwise, generate a new one